### ISSUE TITLE
 EntityDamageByEntityEvent error

### ISSUE THREAD

**Issue by alec-jensen** [kidney-bean/generalutils#3]
https://media.discordapp.net/attachments/1052411680593817701/1061805176656253078/image.png

**Comment by alec-jensen** [comment]
stack trace:
```[23:21:37 ERROR]: Could not pass event EntityDamageByEntityEvent to GeneralUtils v0.1.1
java.lang.ClassCastException: class org.bukkit.craftbukkit.v1_19_R1.entity.CraftPillager cannot be cast to class org.bukkit.entity.Player (org.bukkit.craftbukkit.v1_19_R1.entity.CraftPillager and org.bukkit.entity.Player are in unnamed module of loader java.net.URLClassLoader @484b61fc)
        at me.alecjensen.generalutils.commands.FreezeCommand.EntityDamageByEntityEvent(FreezeCommand.java:83) ~[generalutils-0.1.1-all.jar:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor41.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:77) ~[purpur-api-1.19.2-R0.1-SNAPSHOT.jar:?]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[purpur-api-1.19.2-R0.1-SNAPSHOT.jar:git-Purpur-1858]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[purpur-api-1.19.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:678) ~[purpur-api-1.19.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.callEvent(CraftEventFactory.java:249) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:1173) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:1022) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:973) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at org.bukkit.craftbukkit.v1_19_R1.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:1207) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at net.minecraft.world.entity.LivingEntity.damageEntity0(LivingEntity.java:2205) ~[?:?]
        at net.minecraft.world.entity.LivingEntity.hurt(LivingEntity.java:1424) ~[?:?]
        at net.minecraft.world.entity.raid.Raider.hurt(Raider.java:299) ~[?:?]
        at net.minecraft.world.entity.player.Player.attack(Player.java:1372) ~[?:?]
        at net.minecraft.server.level.ServerPlayer.attack(ServerPlayer.java:2212) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl$6.a(ServerGamePacketListenerImpl.java:3081) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundInteractPacket$1.dispatch(ServerboundInteractPacket.java:24) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundInteractPacket.dispatch(ServerboundInteractPacket.java:80) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handleInteract(ServerGamePacketListenerImpl.java:3000) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundInteractPacket.handle(ServerboundInteractPacket.java:67) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundInteractPacket.handle(ServerboundInteractPacket.java:12) ~[?:?]
        at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:51) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1368) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:185) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1345) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1338) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1316) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1204) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:310) ~[purpur-1.19.2.jar:git-Purpur-1858]
        at java.lang.Thread.run(Thread.java:833) ~[?:?```

**Comment by alec-jensen** [comment]
fixed in [8cb33c6](https://github.com/alec-jensen/generalutils/commit/8cb33c6f292a3cd8afb1f46085207b77e1a9c525)
